naga/tests/out/hlsl/image.hlsl

```diff
@@ -369,5 +369,6 @@ float4 depth_no_comparison() : SV_Target0
     float2 tc_3 = (0.5).xx;
     float s2d_1 = image_2d_depth.Sample(sampler_reg, tc_3);
     float4 s2d_gather = image_2d_depth.Gather(sampler_reg, tc_3);
-    return ((s2d_1).xxxx + s2d_gather);
+    float s2d_level = image_2d_depth.SampleLevel(sampler_reg, tc_3, 1);
```

Comment:
> For HLSL and MSL, `SampleLevel()` and `metal::level()` appear to implicitly convert the integer to floats just fine.

I worry that the way we test it here works because the `1` gets interpreted as a float. If you change the input WGSL to be `var level = 1;` instead of `let level = 1;`, does it still work? That should cause the variable to show up in the output.

Reply:
Ok, if that works then I guess it's fine. I didn't know HLSL and MSL did automatic conversions.